src/storage_utils.cpp

```diff
@@ -174,7 +174,8 @@ namespace libtorrent { namespace aux {
 			if (ec) return -1;
 
 			// advance our position in the iovec array and the file offset.
-			current_buf = advance_bufs(current_buf, bytes_transferred);
```

Comment:
yes, I saw that option too, didn't do it because it involve more code, let me do that